app/app.go

```diff
@@ -602,10 +552,104 @@ func NewAxelarApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
+		BaseApp:           bApp,
+		appCodec:          appCodec,
+		interfaceRegistry: interfaceRegistry,
+		keys:              keys,
+		upgradeKeeper:     upgradeK,
+	}
+
+	mm = module.NewManager(
 		appModules...,
 	)
+	app.mm = mm
+
+	app.mm.SetOrderMigrations(orderMigrations()...)
+	app.mm.SetOrderBeginBlockers(orderBeginBlockers()...)
+	app.mm.SetOrderEndBlockers(orderEndBlockers()...)
+	app.mm.SetOrderInitGenesis(orderModulesForGenesis()...)
+
+	app.mm.RegisterInvariants(&crisisK)
+
+	// register all module routes and module queriers
+	app.mm.RegisterRoutes(app.Router(), app.QueryRouter(), encodingConfig.Amino)
+	configurator = module.NewConfigurator(app.appCodec, app.MsgServiceRouter(), app.GRPCQueryRouter())
+	app.mm.RegisterServices(configurator)
+
+	// initialize stores
+	app.MountKVStores(keys)
+	app.MountTransientStores(tkeys)
+	app.MountMemoryStores(memKeys)
+
+	// The initChainer handles translating the genesis.json file into initial state for the network
+	app.SetInitChainer(app.InitChainer)
+	app.SetBeginBlocker(app.BeginBlocker)
+	app.SetEndBlocker(app.EndBlocker)
+
+	// The baseAnteHandler handles signature verification and transaction pre-processing
+	baseAnteHandler, err := authAnte.NewAnteHandler(
+		authAnte.HandlerOptions{
+			AccountKeeper:   accountK,
+			BankKeeper:      bankK,
+			SignModeHandler: encodingConfig.TxConfig.SignModeHandler(),
+			FeegrantKeeper:  feegrantK,
+			SigGasConsumer:  authAnte.DefaultSigVerificationGasConsumer,
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	anteDecorators := []sdk.AnteDecorator{
+		ante.NewAnteHandlerDecorator(baseAnteHandler),
+	}
+
+	// enforce wasm limits earlier in the ante handler chain
+	if IsWasmEnabled() {
+		anteDecorators = append(anteDecorators, wasmAnteDecorators...)
+	}
+
+	anteDecorators = append(anteDecorators,
+		ante.NewLogMsgDecorator(appCodec),
+		ante.NewCheckCommissionRate(stakingK),
+		ante.NewUndelegateDecorator(multisigK, nexusK, snapK),
+		ante.NewCheckRefundFeeDecorator(app.interfaceRegistry, accountK, stakingK, snapK, rewardK),
+		ante.NewCheckProxy(snapK),
+		ante.NewRestrictedTx(permissionK),
+		ibcante.NewAnteDecorator(ibcKeeper),
+	)
+
+	anteHandler := sdk.ChainAnteDecorators(
+		anteDecorators...,
+	)
+	app.SetAnteHandler(anteHandler)
+
+	if loadLatest {
+		if err := app.LoadLatestVersion(); err != nil {
+			tmos.Exit(err.Error())
+		}
+
+		if IsWasmEnabled() {
+			ctx := app.BaseApp.NewUncachedContext(true, tmproto.Header{})
+
+			// Initialize pinned codes in wasmvm as they are not persisted there
+			if err := wasmK.InitializePinnedCodes(ctx); err != nil {
+				tmos.Exit(fmt.Sprintf("failed initialize pinned codes %s", err))
+			}
+		}
+	}
+
+	/* ==== at this point all stores are fully loaded ==== */
+
+	// we need to ensure that all chain subspaces are loaded at start-up to prevent unexpected consensus failures
+	// when the params keeper is used outside the evm module's context
+	evmK.InitChains(app.NewContext(true, tmproto.Header{}))
```

Comment:
this part is not new, git just has difficulties matching it up with the previous state